storm-core/src/jvm/org/apache/storm/serialization/KryoTupleDeserializer.java

```
@@ -50,7 +54,8 @@ public Tuple deserialize(byte[] ser) {
             List<Object> values = _kryo.deserializeFrom(_kryoInput);
             return new TupleImpl(_context, values, taskId, streamName, id);
         } catch(IOException e) {
-            throw new RuntimeException(e);
+            LOG.error("Failed to deserialize tuple.", e);
```

Then we need to secure the connection against attach and not mask real errors.